src/Diagnostics.DataProviders/DataProviders/MdmDataProvider.cs

```diff
@@ -0,0 +1,185 @@
+        /// </summary>
+        /// <param name="cache">Operation cache</param>
+        /// <param name="configuration">Data provider configuration</param>
+        /// <param name="requestId">Request id.</param>
+        public MdmDataProvider(OperationDataCache cache, MdmDataProviderConfiguration configuration, string requestId) : base(cache)
+        {
+            _configuration = configuration;
+            _mdmClient = MdmClientFactory.GetMdmClient(configuration, requestId);
+        }
+
+        /// <summary>
+        /// Get metadata.
+        /// </summary>
+        /// <returns>Data provider metadata.</returns>
+        public DataProviderMetadata GetMetadata()
+        {
+            return null;
+        }
+
+        /// <summary>
+        /// Gets the list of namespaces for the monitoringAccount.
+        /// </summary>
+        /// <returns>The list of namespaces for the monitoringAccount.</returns>
+        public async Task<IEnumerable<string>> GetNamespacesAsync()
+        {
+            return await _mdmClient.GetNamespacesAsync(_configuration.MonitoringAccount);
+        }
+
+        /// <summary>
+        /// Gets the list of metric names for the monitoringAccount and metricNamespace.
+        /// </summary>
+        /// <param name="metricNamespace">The metric namespace.</param>
+        /// <returns>The list of metric names for the monitoringAccount and metricNamespace.</returns>
+        public async Task<IEnumerable<string>> GetMetricNamesAsync(string metricNamespace)
+        {
+            return await _mdmClient.GetMetricNamesAsync(_configuration.MonitoringAccount, metricNamespace);
+        }
+
+        /// <summary>
+        /// Gets the list of dimension names for the metricId.
+        /// </summary>
+        /// <param name="metricNamespace">Metric namespace</param>
+        /// <param name="metricName">Metric name</param>
+        /// <returns>The list of dimension names for the metricId.</returns>
+        public async Task<IEnumerable<string>> GetDimensionNamesAsync(string metricNamespace, string metricName)
+        {
+            return await _mdmClient.GetDimensionNamesAsync(_configuration.MonitoringAccount, metricNamespace, metricName);
+        }
+
+        /// <summary>
+        /// Gets the dimension values for dimensionName satifying the dimensionFilters and
+        /// </summary>
+        /// <param name="metricNamespace">Metric namespace</param>
+        /// <param name="metricName">Metric name</param>
+        /// <param name="filter">The dimension filters representing the pre-aggregate dimensions. Create an emtpy include filter for dimension with no filter values. Requested dimension should also be part of this and should be empty.</param>
+        /// <param name="dimensionName">Name of the dimension for which values are requested.</param>
+        /// <param name="startTimeUtc">Start time for evaluating dimension values.</param>
```

Comment:
Don't make the startTime and endTime change default just yet, I didn't realize that we have to make changes in other parts of the project before each data provider can inherit the starttime and endtime from the UI. Let's just get this out and then I can talk to you afterwards on how to inherit the startTime and endTime

Reply:
Okay, I will keep the change.